titus-ext/cassandra/src/main/java/com/netflix/titus/ext/cassandra/store/CassandraLoadBalancerStore.java

```diff
@@ -86,30 +106,49 @@
      */
     private final ConcurrentMap<String, SortedSet<JobLoadBalancer>> jobToAssociatedLoadBalancersMap;
 
-    private static final String GET_ALL_JOB_IDS = String
+    private static final String GET_ALL_ASSOCIATIONS = String
             .format("SELECT %s, %s, %s FROM %s;",
                     COLUMN_JOB_ID,
                     COLUMN_LOAD_BALANCER,
                     COLUMN_STATE,
-                    TABLE_LOAD_BALANCER);
-    private static final String INSERT_JOB_LOAD_BALANCER = String
+                    TABLE_LOAD_BALANCER_ASSOCIATIONS);
+    private static final String INSERT_ASSOCIATION = String
             .format("INSERT INTO %s(%s, %s, %s) VALUES (?, ?, ?);",
-                    TABLE_LOAD_BALANCER,
+                    TABLE_LOAD_BALANCER_ASSOCIATIONS,
                     COLUMN_JOB_ID,
                     COLUMN_LOAD_BALANCER,
                     COLUMN_STATE);
-    private static final String UPDATE_JOB_LOAD_BALANCER_STATE = String
-            .format("UPDATE %s SET %s = ? WHERE %s = ? AND %s = ?;",
-                    TABLE_LOAD_BALANCER,
-                    COLUMN_STATE,
-                    COLUMN_JOB_ID,
-                    COLUMN_LOAD_BALANCER);
-    private static final String DELETE_JOB_LOAD_BALANCER = String
+    private static final String DELETE_ASSOCIATION = String
             .format("DELETE FROM %s WHERE %s = ? AND %s = ?",
-                    TABLE_LOAD_BALANCER,
+                    TABLE_LOAD_BALANCER_ASSOCIATIONS,
                     COLUMN_JOB_ID,
                     COLUMN_LOAD_BALANCER);
 
+    private static Pair<JobLoadBalancer, JobLoadBalancer.State> buildLoadBalancerStatePairFromRow(Row row) {
+        String jobId = row.getString(COLUMN_JOB_ID);
+        String loadBalancerId = row.getString(COLUMN_LOAD_BALANCER);
+        String state = row.getString(COLUMN_STATE);
+        JobLoadBalancer.State parsedState;
+        if (state == null) {
+            logger.warn("Unexpected null state for association {}:{}", jobId, loadBalancerId);
+            parsedState = JobLoadBalancer.State.DISSOCIATED;
+        } else {
+            parsedState = JobLoadBalancer.State.valueOf(state.toUpperCase());
```

Comment:
Try/cach in case the value is corrupted? Unless this error would be handled somewhere else.

Reply:
I think we should let the error bubble up and prevent activation here, since this would be a case of data corruption. FWIW the code previous to changes here was doing the same (only without `.toUpperCase()`).